File: frontend/src/mateu/shared/apiClients/dtos/UI.ts
import Menu from "./Menu";
import App from "./App";

export default interface UI {

    favIcon: string | undefined
    logo: string | undefined
    title: string | undefined
    subtitle: string | undefined
    menu: Menu[]  | undefined
    homeJourneyTypeId: string | undefined
    loginUrl: string | undefined
    logoutUrl: string | undefined
    apps: App[] | undefined

}

File: backend/webflux/dtos/src/main/java/io/mateu/dtos/UI.java
package io.mateu.dtos;

import java.util.Collections;
import java.util.List;
import lombok.*;

public record UI(
    String favIcon,
    String icon,
    String logo,
    String title,
    String subtitle,
    List<Menu> menu,
    String homeJourneyTypeId,
    String loginUrl,
    String logoutUrl,
    List<App> apps) {

  public UI {
    menu = Collections.unmodifiableList(menu);
    apps = Collections.unmodifiableList(apps);
  }

  @Override
  public List<Menu> menu() {
    return Collections.unmodifiableList(menu);
  }

  @Override
  public List<App> apps() {
    return Collections.unmodifiableList(apps);
  }
}
